### PR TITLE
[FIX] intrastat_product - set region_code based on code or the region

### PR DESCRIPTION
**Pull request by ghost**
If we don't use the method "_get_region_code" and inherit the method in the localisation module, the region code is never set even when using the default method "_get_region" that fetchs the region based on invoice, warehouse or company.

### CODE REVIEW

File: intrastat_product/models/intrastat_product_declaration.py [hunk 1]
```diff
@@ -709,7 +709,7 @@ def _gather_invoices(self, notedict):
                     "amount_accessory_cost_company_currency": 0.0,
                     "transaction_id": intrastat_transaction.id,
                     "product_origin_country_id": product_origin_country.id or False,
-                    "region_code": region_code,
+                    "region_code": region_code or region and region.code or False,
```
Comment (pedrobaeza) [inline intrastat_product/models/intrastat_product_declaration.py]: Please avoid this old-stylish spaghetti code:

```suggestion
                    "region_code": region_code or region.code,
```

Comment (pedrobaeza) [reply intrastat_product/models/intrastat_product_declaration.py]: If `region` may be False, then the method `_get_region` should return an empty recordset.

Comment (ghost) [reply intrastat_product/models/intrastat_product_declaration.py]: @pedrobaeza Code adapted according to your suggestion. Thanks

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
It seems one test is not satisfied. Please check.

**Comment by ghost** [comment]
@pedrobaeza The issue is fixed

**Pull request merged**